net/dns/quad100.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package dns
+
+import (
+	"bytes"
+	"context"
+	"net"
+	"net/netip"
+	"time"
+)
+
+type ManagerConn struct {
+	Ctx        context.Context
```

Comment:
do we need to curry a context? whose is this? what's the scope/how's it used? It doesn't unblock Reads, for instance.

Reply:
It is the context provided to userDialResolve and is used for querying the DNS Manager. There was an extra context passed to the Quad100ResolverDial function which didn't seem useful, I removed it